src/symbols.c

```diff
@@ -1914,27 +1867,276 @@ static void load_user_tags(filetype_id ft_id)
 }
 
 
+static void on_focus_out(GtkWidget *list, GdkEventFocus *unused, gpointer *user_data)
+{
+	gtk_widget_hide(tag_goto_popup);
+}
+
+
+static void on_row_activated(GtkTreeView *tree_view, GtkTreePath *path,
+	GtkTreeViewColumn *column, gpointer user_data)
+{
+	GtkTreeModel *model = gtk_tree_view_get_model(tree_view);
+	GeanyDocument *new_doc;
+	GtkTreeIter iter;
+	TMTag *tag;
+
+	gtk_tree_model_get_iter(model, &iter, path);
+	gtk_tree_model_get(model, &iter, TAG_COLUMN, &tag, -1);
+	g_return_if_fail(tag);
+
+	new_doc = document_find_by_real_path(tag->file->file_name);
+	if (!new_doc)
+		new_doc = document_open_file(tag->file->file_name, FALSE, NULL, NULL);
+
+	if (new_doc)
+	{
+		GeanyDocument *old_doc = document_get_current();
+
+		navqueue_goto_line(old_doc, new_doc, tag->line);
+	}
+
+	gtk_widget_hide(tag_goto_popup);
```

Comment:
the whole store should probably be cleared when hiding, not to keep TMTags around for no reason.  note: destroying the popup would take care of this.